core/trino-main/src/main/java/io/trino/metadata/PolymorphicScalarFunctionBuilder.java

```diff
@@ -278,7 +278,7 @@ public ChoiceBuilder implementation(Function<MethodsGroupBuilder, MethodsGroupBu
                 argumentConventions = nCopies(signature.getArgumentTypes().size(), NEVER_NULL);
             }
             MethodsGroupBuilder methodsGroupBuilder = new MethodsGroupBuilder(clazz, signature, argumentConventions);
-            methodsGroupSpecification.apply(methodsGroupBuilder);
+            methodsGroupBuilder = methodsGroupSpecification.apply(methodsGroupBuilder);
```

Comment:
I skimmed the usages and it seems `methodsGroupSpecification` could be a `Consumer`.

Reply:
I actually think that what I did here is the better way, I mean it's more like how the builder pattern works internally. But I can change it into a `Consumer`as well.